server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/RollbackPlanRuntimeFlowableTaskCopied.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    protected void onSuccessful(Long taskId, TaskService taskService) {
+        log.info("Generate rollback plan task succeed, taskId={}", taskId);
+    }
+
+    @Override
+    protected void onTimeout(Long taskId, TaskService taskService) {}
+
+    @Override
+    protected void onProgressUpdate(Long taskId, TaskService taskService) {}
+
+    @Override
+    protected boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
+        return false;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return false;
+    }
+
+    private JobDefinition buildJobDefinition(@NonNull TaskEntity entity, @NonNull ConnectionConfig config) {
+        Map<String, String> jobParams = new HashMap<>();
+        jobParams.put(JobParametersKeyConstants.TASK_PARAMETER_JSON_KEY,
+                JobUtils.toJson(buildRollbackPlanTaskParameters(entity, config)));
+        jobParams.put(JobParametersKeyConstants.TASK_EXECUTION_TIMEOUT_MILLIS,
+                String.valueOf(rollbackProperties.getMaxTimeoutMillisecond()));
+        return DefaultJobDefinition.builder()
+                .jobClass(RollbackPlanTask.class)
+                .jobType(TaskType.GENERATE_ROLLBACK.name())
+                .jobParameters(jobParams)
+                .build();
+    }
+
+    private RollbackPlanTaskParameters buildRollbackPlanTaskParameters(@NonNull TaskEntity entity,
+            @NonNull ConnectionConfig config) {
+        RollbackPlanTaskParameters rollbackPlanTaskParameters = new RollbackPlanTaskParameters();
+        rollbackPlanTaskParameters.setFlowInstanceId(getFlowInstanceId());
+        rollbackPlanTaskParameters.setRollbackProperties(copyRollbackPropertiesBean());
+        rollbackPlanTaskParameters.setConnectionConfig(config);
+        rollbackPlanTaskParameters.setDefaultSchema(entity.getDatabaseName());
+        DatabaseChangeParameters p = JsonUtils.fromJson(entity.getParametersJson(), DatabaseChangeParameters.class);
+        rollbackPlanTaskParameters.setSqlContent(p.getSqlContent());
+        if (CollectionUtils.isNotEmpty(p.getSqlObjectIds())) {
+            List<ObjectMetadata> objectMetadatas = new ArrayList<>();
+            String bucket = "async".concat(File.separator).concat(String.valueOf(entity.getCreatorId()));
+            for (String objectId : p.getSqlObjectIds()) {
+                ObjectMetadata metadata = objectStorageFacade.loadMetaData(bucket, objectId);
+                objectMetadatas.add(metadata);
+            }
+            rollbackPlanTaskParameters.setSqlFileObjectMetadatas(objectMetadatas);
+        }
+        rollbackPlanTaskParameters.setDelimiter(p.getDelimiter());
+        return rollbackPlanTaskParameters;
+    }
+
+    /**
+     * Copy the rollback properties bean in case GSON cannot serialize Spring bean
+     * 
+     * @return A new rollback properties simple Java object
+     */
+    private RollbackProperties copyRollbackPropertiesBean() {
```

Comment:
reference ObjectUtil#deepCopy

Reply:
The `RollbackProperties` is not a simple Java object but a Spring proxy bean. It cannot be deep copied by `ObjectUtil#deepCopy`.

Reply:
use `RollbackProperties rollbackProperties = new RollbackProperties()` is normal bean, here should be changed if  RollbackProperties add property, 
btw: 
why need copy a new bean in `rollbackPlanTaskParameters.setRollbackProperties(copyRollbackPropertiesBean())`;

Reply:
The autowired `RollbackProperties` is a Spring proxy bean, which cannot be serialized by GSON. So I `new` a new Java object for this class. I have try to use `ObjectUtil#deepCopy`, but this method return because this method depends on JsonUtil and Fastjson cannot serialize this Spring bean. So the right and easiest solution is new a object and copy properties.